packages/palette/src/utils/useOnScroll.ts

```diff
@@ -7,21 +7,26 @@ export const useOnScroll = (ref: RefObject<HTMLElement>) => {
   const [isScrolled, setIsScrolled] = useState(false)
 
   useEffect(() => {
+    if (!("IntersectionObserver" in window)) return
+
     const node = ref.current
-    if (node) {
-      const onScroll = () => {
-        if (node.scrollTop > 0) {
-          !isScrolled && setIsScrolled(true)
-        } else {
-          isScrolled && setIsScrolled(false)
-        }
+    if (!node) return
+
+    const observer = new IntersectionObserver(
+      ([entry]) => {
+        setIsScrolled(!entry.isIntersecting)
+      },
+      {
+        root: null,
+        rootMargin: "0px",
+        threshold: 1.0,
       }
+    )
 
-      node.addEventListener("scroll", onScroll, false)
+    observer.observe(ref.current)
 
-      return () => {
-        node.removeEventListener("scroll", onScroll, false)
-      }
+    return () => {
+      observer.unobserve(node)
```

Comment:
I'd also set up and return the ref inside of the hook so that in the call site you just have to pull out the ref and place it where you want to track.

Reply:
If I got you correctly, this [commit](https://github.com/artsy/palette/pull/1076/commits/34a6633ae6e1bde716fdd80d2d182527cc0c6951) does what you asked 😄

Reply:
Ah lol thanks